allow github concourse contributor team to rerun builds

xtremerui opened the pull request:
once we have the pr bot working by https://github.com/concourse/concourse/pull/5564

then we need to config concourse production CI to allow members of concourse:contributor team to rerun failure PR pipeline builds(due to flaky test) and check PR resource(if desired version is not fetched)

Commit:
allow github concourse contributor team to rerun builds

and check resources so they could retrigger failure build on PR
pipeline and check pr resource in case wanted version is not
available.

Note: they could also rerun any builds and check any resources by
this change.

Signed-off-by: Rui Yang <ryang@pivotal.io>

jamieklassen commented on deployments/with-creds/ci/values.yaml:
```diff
@@ -82,11 +82,20 @@ concourse:
             team: concourse:Pivotal
         github:
           enabled: true
+      configRBAC: |
+        pipeline-operator:
+        - RerunJobBuild
+        - CheckResource
```
I'm pretty sure this is a no-op, since [these](https://github.com/concourse/concourse/blob/cb9530ecf62340c4fa5c17e14ec2f1a72d0c6b0d/atc/api/accessor/roles.go#L29) [actions](https://github.com/concourse/concourse/blob/cb9530ecf62340c4fa5c17e14ec2f1a72d0c6b0d/atc/api/accessor/roles.go#L48) are already assigned to the pipeline-operator role.

If you want to restrict pipeline operators so that they can _only_ perform these actions (like no pinning/pausing/aborting/enabling/disabling) I think you want to promote those actions to `member`:
```suggestion
      configRBAC: |
        member:
        - AbortBuild
        - CreateJobBuild
        - PauseJob
        - UnpauseJob
        - ClearTaskCache
        - UnpinResource
        - SetPinCommentOnResource
        - CheckResourceWebHook
        - CheckResourceType
        - EnableResourceVersion
        - DisableResourceVersion
        - PinResourceVersion
        - PausePipeline
        - UnpausePipeline
```
or else you could grant these github team members the `viewer` role and demote rerunning and checking so that viewers can do it:
```suggestion
      configRBAC: |
        viewer:
        - RerunJobBuild
        - CheckResource
```

jamieklassen commented on deployments/with-creds/ci/values.yaml:
```diff
@@ -82,11 +82,20 @@ concourse:
         mainTeam:
+          config: |
+            roles:
+            - name: pipeline-operator
```
see comment below - if you decide to take the second approach, you'll want this to be ~`member`~ `viewer`. Otherwise it's fine.

xtremerui replied:
If taking second approach, which is demote those two permission to viewer, then the change here should be `viewer` right? 

If taking first approach, then `pipeline-operator` will have only two permissions left, then we don't need to update here.

Just want to confirm.

jamieklassen replied:
yeah, my bad

jamieklassen reviewed (changes requested):
It seems like `--config-rbac` is pretty weird. I'd guess you have encountered the same confusion as @gowrisankar22 in concourse/concourse#5514.

xtremerui commented:
@pivotal-jamie-klassen we realize giving the permission of `RerunJobBuild` to contributor actually endanger some critical jobs we have in CI main team for example those `ship-it` jobs in all release pipelines.

So we are thinking moving `PRs` pipeline to a different concourse team like `contributors-team` and use fly set-team after concourse deployment to give contributors pipeline-operators role. In this way the `RerunJobBuild` permission wil be limited. The con is its a manual process and we need to consider this for green peace for automation.

WDYT?

cirocosta commented:
> [...] use `fly set-team` after concourse deployment to give contributors pipeline-operators role. 
> [...] a manual process and we need to consider this for green peace for automation.
 
that just reminded me of [RFC: Concourse k8s operator](https://github.com/concourse/rfcs/pull/45) and the [Pipeline CRD RFC](https://github.com/concourse/rfcs/pull/44) - with a `Team` CRD, we could have the configuration as code "and *voila*

update: and without the use of concourse tokens, having client auth grant type being a thing, we could soon have our controller doing all of that stuff without being admin :eyes:

aoldershaw commented:
@cirocosta I know CRDs are the hot thing, but what if we had a Concourse terraform provider instead? Where you could define teams/pipelines using HCL, and when you spin up Concourse, you could have your initial teams/pipelines created as well all through Terraform?

There may be other benefits in using CRD+operator would bring over a Terraform provider, though - what do you think? It's possible we'd be appealing to a wider demographic in K8s than Terraform, I suppose

jamieklassen commented:
@cirocosta @aoldershaw rather than investing in a third-party plugin for team automation, I would vote for a core concourse feature, a la `set_team` step: https://github.com/concourse/rfcs/discussions/50. This could perhaps be wrapped in a k8s operator or terraform provider.

xtremerui commented:
Updated to config RBAC in CI deployment only. Added a team config file for manual set-team step for now. Also created `contributor` team in CI. Next step will be move PRs pipeline to this team.

jamieklassen reviewed (approved):
Looks good!

Commit:
add owner to contributor team config

Signed-off-by: Rui Yang <ryang@pivotal.io>

xtremerui commented:
@pivotal-jamie-klassen I forgot to put concourse:pivotal as owner for team contributor. Seems we are doing this for all other teams on CI.

jamieklassen commented:
Should be ok since concourse:pivotal are super admins
